### PR TITLE
[VL-67] - Add timestamp and traceId to error response

### PR DESCRIPTION
**Pull request by kin0992**
#### List of Changes

Add missing parameters on 400 response body.

#### Motivation and Context

The OpenAPI is not aligned with the implementation of the system, so the emulator doesn't behave like PN.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/adapters/http/Problem.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ export const fromErrors = (errors: ts.Errors): Problem => ({
   ...fromNumber(400),
   title: 'Bad Request',
   detail: PR.failure(errors),
+  timestamp: new Date(),
+  traceId: 'aTraceId',
```
Comment (datalek) [inline src/adapters/http/Problem.ts:10]: I think we should keep the format provided by PN
```suggestion
  traceId: 'Self=1-631b10db-2c571f3f79954187229939e5;Root=1-631b11db-39caf6636f4ed15618461d95;Parent=6e2ef00e703981ac;Sampled=1,
```

**Review by datalek — CHANGES_REQUESTED** [review]
The emulator returns an error with the field `errors` always empty, while the real PN returns detailed information about the error (e.g. The missing field). What do you think to try to replicate the same structure provided by PN?

p.s.: At the moment the detailed information are not described into the open-api specification.

File: src/adapters/http/Problem.ts (hunk 1 above)
Comment (kin0992) [reply src/adapters/http/Problem.ts]: Replaced with the correct value

**Comment by kin0992** [comment]
> The emulator returns an error with the field `errors` always empty, while the real PN returns detailed information about the error (e.g. The missing field). What do you think to try to replicate the same structure provided by PN?
> 
> p.s.: At the moment the detailed information are not described into the open-api specification.

Generally speaking, I agree with you, but in this case, we are discovering a lot of discrepancies with the OpenAPI (and this is not something new). Following your suggestion (which makes sense to me), I'm asking you this: do you also think we have to update the OpenAPI specification (update the forked repository) so that the `type` field is no longer required since the server doesn't return it in the response body?
My question aims to think about if those changes are strictly necessary at this moment or not.

**Comment by datalek** [comment]
> > The emulator returns an error with the field `errors` always empty, while the real PN returns detailed information about the error (e.g. The missing field). What do you think to try to replicate the same structure provided by PN?
> > p.s.: At the moment the detailed information are not described into the open-api specification.
> 
> Generally speaking, I agree with you, but in this case, we are discovering a lot of discrepancies with the OpenAPI (and this is not something new). Following your suggestion (which makes sense to me), I'm asking you this: do you also think we have to update the OpenAPI specification (update the forked repository) so that the `type` field is no longer required since the server doesn't return it in the response body? My question aims to think about if those changes are strictly necessary at this moment or not.

Good point. If we don't want to align the open-api definition, we could leave the issue card open or change it to indicate that not all of the discrepancies have been resolved because of the unaligned open-api.

**Comment by kin0992** [comment]
> > > The emulator returns an error with the field `errors` always empty, while the real PN returns detailed information about the error (e.g. The missing field). What do you think to try to replicate the same structure provided by PN?
> > > p.s.: At the moment the detailed information are not described into the open-api specification.
> > 
> > 
> > Generally speaking, I agree with you, but in this case, we are discovering a lot of discrepancies with the OpenAPI (and this is not something new). Following your suggestion (which makes sense to me), I'm asking you this: do you also think we have to update the OpenAPI specification (update the forked repository) so that the `type` field is no longer required since the server doesn't return it in the response body? My question aims to think about if those changes are strictly necessary at this moment or not.
> 
> Good point. If we don't want to align the open-api definition, we could leave the issue card open or change it to indicate that not all of the discrepancies have been resolved because of the unaligned open-api.

I added a point to our weekly meeting, so we can discuss there on how we should handle this situation.

**Review by datalek — APPROVED** [review]
💥

**Pull request merged**